File: 217.contains-duplicate.go
/*
 * @lc app=leetcode id=217 lang=golang
 *
 * [217] Contains Duplicate
 */

// @lc code=start
func containsDuplicate(nums []int) bool {
    m := make(map[int]bool)
	for _, num := range nums {
		if m[num] {
			return true
		}
		m[num] = true
	}
	return false
}
// @lc code=end

File: 217.contains-duplicate.rs
/*
 * @lc app=leetcode id=217 lang=rust
 *
 * [217] Contains Duplicate
 */

// @lc code=start[]
impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut m = std::collections::HashSet::new();
        !nums.iter().all(|n| m.insert(n))
    }
}
// @lc code=end

